### DISCUSSION TITLE
How to listen for client-side route changes

### DISCUSSION THREAD

**Discussion by amumufun** [vikejs/vike#1766] [Discussion]
Similar to vue-router's beforeEach

**Comment by brillout** [comment]
https://vike.dev/onRenderClient#see-also

It seems like you could have find this yourself? Have a look at https://brillout.github.io/rules/#try.

**Reply by amumufun** [reply]
I want to monitor the route changes within the page, similar to vue-router's beforeRouteUpdate, to load page data.

**Reply by brillout** [reply]
> monitor [...] route change

I don't know your specific requirements here, but it seems like there are more than enough hooks to achieve what you want.

That's why I linked you to https://brillout.github.io/rules/#try as it's usually best to follow these rules on GitHub. See also https://vike.dev/faq#how-can-i-reach-out-for-help.

That said, feel free to elaborate your goal if you didn't manage to achieve it.

**Reply by brillout** [reply]
I guess this is what you were looking for: https://github.com/vikejs/vike/discussions/1781#discussioncomment-10207610. Do you mind elaborating why these this works for you whereas using Vike hooks doesn't?
